kaldb/src/test/java/com/slack/kaldb/testlib/MetricsUtil.java

```diff
@@ -1,17 +1,66 @@
 package com.slack.kaldb.testlib;
 
 import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.search.MeterNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+// The get*() functions do some exception handling.
+// most likely we'll be calling getCount from an await() and waiting for the counter to match
+// expected value. When the meter has not been initialized we want await() to actually wait
+// till the code initializes the meter.
+// If we don't add a catch block we'll throw MeterNotFoundException and await() code needs
+// special handling to ensure the exception is dealt with
+// so instead of every called doing something like this below we just return 0
+//      await()
+//              .until(
+//                      () -> {
+//                        try {
+//                          double count = getCount(METRIC_NAME, meterRegistry);
+//                          LOG.debug("METRIC_NAME current_count={} total_count={}", count,
+// meterRegistry);
+//                          return count == <count_expected>;
+//                        } catch (MeterNotFoundException e) {
+//                          return false;
+//                        }
+//                      });
 public class MetricsUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MetricsUtil.class);
+
   public static double getCount(String counterName, MeterRegistry metricsRegistry) {
-    return metricsRegistry.get(counterName).counter().count();
+    try {
+      return metricsRegistry.get(counterName).counter().count();
+    } catch (MeterNotFoundException e) {
+      LOG.error("Metric not found", e);
```

Comment:
This can be an info/warn saying we the meter is not found so returning 0. This is not an error since we continue the method execution.

Does the exception have the meter name? If not, can we also log it here?

Reply:
changed to warn. Meter name is printed. additionally the stack trace usually makes it very evident which test line caused this and also infer the metric being referenced.